c/tskit/tables.c

```diff
@@ -8231,49 +8284,10 @@ tsk_table_collection_union(tsk_table_collection_t *self, tsk_table_collection_t
         if (other_node_mapping[k] != TSK_NULL) {
             node_map[k] = other_node_mapping[k];
         } else {
-            ret = tsk_node_table_get_row(&other->nodes, k, &node);
-            if (ret < 0) {
-                goto out;
-            }
-            new_ind = TSK_NULL;
-            if (node.individual != TSK_NULL) {
-                if (individual_map[node.individual] == TSK_NULL) {
-                    tsk_individual_table_get_row(
-                        &other->individuals, node.individual, &ind);
-                    ret = tsk_individual_table_add_row(&self->individuals, ind.flags,
-                        ind.location, ind.location_length, ind.metadata,
-                        ind.metadata_length);
-                    if (ret < 0) {
-                        goto out;
-                    }
-                    individual_map[node.individual] = ret;
-                }
-                new_ind = individual_map[node.individual];
-            }
-            new_pop = TSK_NULL;
-            if (node.population != TSK_NULL) {
-                // keep same pops if add_populations is False
-                if (!add_populations) {
-                    population_map[node.population] = node.population;
-                }
-                if (population_map[node.population] == TSK_NULL) {
-                    tsk_population_table_get_row(
-                        &other->populations, node.population, &pop);
-                    ret = tsk_population_table_add_row(
-                        &self->populations, pop.metadata, pop.metadata_length);
-                    if (ret < 0) {
-                        goto out;
-                    }
-                    population_map[node.population] = ret;
-                }
-                new_pop = population_map[node.population];
-            }
-            ret = tsk_node_table_add_row(&self->nodes, node.flags, node.time, new_pop,
-                new_ind, node.metadata, node.metadata_length);
-            if (ret < 0) {
+            ret = tsk_table_collection_add_and_remap_node(self, other, k, individual_map, population_map, node_map, add_populations);
+            if (ret != 0) {
```

Comment:
yep - changing this to `if (ret < 0) {` fixes the errors.

Reply:
yay!